File: src/main/java/no/nav/veilarbvedtaksstotte/client/dokument/ProduserDokumentV2DTO.java
package no.nav.veilarbvedtaksstotte.client.dokument;

import lombok.Data;
import lombok.experimental.Accessors;

import java.util.List;

@Data
@Accessors(chain = true)
public class ProduserDokumentV2DTO {
    String brukerFnr;
    MalType malType;
    String enhetId;
    String begrunnelse;
    List<String> opplysninger;
    boolean utkast;

    public ProduserDokumentV2DTO setBrukerFnr(String brukerFnr) {
        this.brukerFnr = brukerFnr;
        return this;
    }

    public ProduserDokumentV2DTO setMalType(MalType malType) {
        this.malType = malType;
        return this;
    }

    public ProduserDokumentV2DTO setEnhetId(String enhetId) {
        this.enhetId = enhetId;
        return this;
    }

    public ProduserDokumentV2DTO setBegrunnelse(String begrunnelse) {
        this.begrunnelse = begrunnelse;
        return this;
    }

    public ProduserDokumentV2DTO setOpplysninger(List<String> opplysninger) {
        this.opplysninger = opplysninger;
        return this;
    }

    public ProduserDokumentV2DTO setUtkast(boolean utkast) {
        this.utkast = utkast;
        return this;
    }
}

File: src/main/java/no/nav/veilarbvedtaksstotte/client/dokument/ProduserDokumentV2DTO.kt
package no.nav.veilarbvedtaksstotte.client.dokument

import no.nav.common.types.identer.EnhetId
import no.nav.common.types.identer.Fnr

data class ProduserDokumentV2DTO(
    val brukerFnr: Fnr,
    val navn: String,
    val malType: MalType,
    val enhetId: EnhetId,
    val begrunnelse: String?,
    val opplysninger: List<String>,
    val utkast: Boolean,
    val adresse: AdresseDTO
) {
    data class AdresseDTO(
        val adresselinje1: String,
        val adresselinje2: String?,
        val adresselinje3: String?,
        val postnummer: String?,
        val poststed: String?,
        val land: String?
    )
}
